### DISCUSSION TITLE
Link existing database connection instead of creating a whole new database container

### DISCUSSION THREAD

**Discussion by doncezart** [coollabsio/coolify#4706] [Feature Requests]
Currently, when installing Coolify, it automatically makes a postgresql container to serve as the app's database. Some of us have databases already installed and, me for example, I have a centralized database container which hosts databases for all my apps. It would be nice to have to ability to just link an existing database to the app in order to save resources (2 databases under the same postgres instance eats less resources than 2 separate postgres instances)

**Comment by Potapy4** [comment]
+1

**Comment by douwepausma** [comment]
This would be really nice. I've tried editing the compose file, but Coolify will re-add the postgres service whenever I try and deploy it.

**Comment by jcabreraSerrao** [comment]
I had the same problem. My solution was to generate a Docker-Compose and associate it with a network. With that, associate the application with that network. With that, I was able to solve it.

**Comment by mariokellner** [comment]
Sorry for bringing it up but I want to add some information to that topic as well.

In terms of connection details you can use your own credentials in the .env-file within `/data/coolify/source`. If you use another path you could lose your ability to autoupdate, because coolify´s updater script is hardcoded to `/data/coolify/source`.

**Example:**
``` .ENV 
DB_DATABASE=coolify
DB_USERNAME=sql-user-coolify
DB_PASSWORD=supersecretkeyphrase123!
DB_HOST=someinstance-v16
DB_CONNECTION=pgsql
DB_PORT=5432
```

If you want to disable the postgres container completely you could add a file named [`docker-compose.custom.yml`](https://github.com/coollabsio/coolify/blob/e1f940dc229cd10538dac0e6f8d58b127cf424bc/scripts/upgrade.sh#L67) to the path `/data/coolify/source`.

In there you can assign a `donotstart` profile to it that overwrites the container completely:
``` docker-compose.yaml
services:
  coolify:
    depends_on: !override
      - soketi
      - redis

  postgres:
    profiles:
      - donotstart
```

You have to alter the field `depends_on` because it still expects "postgres" as a healthy service. This example is a bit dirty because it removes the check completely rather than pointing them to the central service...

> [!IMPORTANT]
> If you do that, you will **completely lose** the ability to backup the coolify database over the ui. But it gives you complete control over compose structure, which is, in some edge cases more important to have, than the benefit of "just run the installscript".

**Edit:** 
!override within docker-compose of avaliable for [2.24.4](https://docs.docker.com/reference/compose-file/merge/#replace-value) and later!

**Comment by sovetski** [comment]
any news?

**Reply by Cinzya** [reply]
This should already be possible from what mario described above already.

**Reply by sovetski** [reply]
thanks but I found a better and simpler solution for me

**Reply by doncezart** [reply]
Can you get into more details on your simpler solution?
